scene/gui/layout_container.h

```diff
@@ -0,0 +1,122 @@
+/**************************************************************************/
+/*  layout_container.h                                                    */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef LAYOUT_CONTAINER_H
+#define LAYOUT_CONTAINER_H
+
+#include "core/math/math_defs.h"
+#include "scene/gui/container.h"
+
+class LayoutContainer : public Container {
+	GDCLASS(LayoutContainer, Container);
+
+public:
+	enum LayoutContainerPreset {
+		PRESET_TOP_LEFT = 0,
+		PRESET_TOP_RIGHT,
+		PRESET_BOTTOM_LEFT,
+		PRESET_BOTTOM_RIGHT,
+		PRESET_CENTER_LEFT,
+		PRESET_CENTER_TOP,
+		PRESET_CENTER_RIGHT,
+		PRESET_CENTER_BOTTOM,
+		PRESET_CENTER,
+		PRESET_LEFT_WIDE,
+		PRESET_TOP_WIDE,
+		PRESET_RIGHT_WIDE,
+		PRESET_BOTTOM_WIDE,
+		PRESET_VCENTER_WIDE,
+		PRESET_HCENTER_WIDE,
+		PRESET_FULL_RECT,
+		PRESET_CUSTOM,
+		PRESET_MAX,
+	};
+
+	const real_t relative_margins_presets[PRESET_CUSTOM][4] = {
```

Comment:
```suggestion
	static const real_t relative_margins_presets[PRESET_CUSTOM][4] = {
```
Makes it shared by all instances.